src/openrct2/rct2/S6Importer.cpp

```diff
@@ -1382,6 +1386,17 @@ class S6Importer final : public IParkImporter
         }
 
         dst->TrackLocation = { src->track_x, src->track_y, src->track_z };
+        if (dst->track_type && src->GetTrackType() == TrackElemType::RotationControlToggleAlias)
```

Comment:
What are you checking `dst->track_type` for?

Reply:
It's due to the boat code above.

Reply:
Well, that would need a comment at least, and an explicit `!= 0` check.

Reply:
Making small diffs, since it seemed to be an issue earlier. It should belong in the true case of the above `if` statement.